### PR TITLE
Add a Travis build

### PR DESCRIPTION
**Pull request by yallop**

### CODE REVIEW

**Comment by hannesm** [comment]
(I just enabled the travis ci hook for this repo... /me providing free janitor services :)

**Comment by yallop** [comment]
Thanks, @hannesm!

File: .travis-ci.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+wget https://raw.githubusercontent.com/ocaml/ocaml-travisci-skeleton/master/.travis-ocaml.sh
+sh .travis-ocaml.sh
+
+eval `opam config env`
+
+opam pin add --yes -n $(pwd)
+opam install --yes depext
+opam depext libnqsb-tls
+opam install --deps-only libnqsb-tls
```
Comment (hannesm) [inline .travis-ci.sh]: here's a `--yes` missing... 1.1 is stuck forever

File: .travis.yml [hunk 2]
```diff
@@ -5,4 +5,4 @@ os:
   - linux
   - osx
 env:
-  - OCAML_VERSION=4.02
\ No newline at end of file
+  - OCAML_VERSION=4.02
```
Comment (hannesm) [inline .travis.yml:8]: would 4.03 work as well? :)

Comment (yallop) [inline .travis.yml:8]: Let's see (be1b632).

File: .travis-ci.sh (hunk 1 above)
Comment (yallop) [inline .travis-ci.sh]: Thanks!  Fixed here: a7c4a12.

File: .travis.yml (hunk 2 above)
Comment (hannesm) [inline .travis.yml:8]: looks like that worked out... :)

**Comment by yallop** [comment]
The `x509` tests are [failing on OS X](https://travis-ci.org/mirleft/libnqsb-tls/jobs/137247318#L3131):

```
### stderr ###
# Fatal error: exception Unix.Unix_error(Unix.EMFILE, "open", "./tests/testcertificates/intermediate/second/second-unknown.pem")
# W: Test 'it' fails: Command '/Users/travis/.opam/4.02.3/build/x509.0.5.2/_build/tests/unittestrunner.native' terminated with error code 2
# E: Failure("Tests had a 100.00% failure rate")
# make: *** [test] Error 1
```

I'll remove `--build-test` from the final `opam install` command in `.travis-ci.sh`

**Comment by yallop** [comment]
There's a fix for the x509 failure here: https://github.com/mirleft/ocaml-x509/pull/83

It may be best to merge that PR then enable `--build-test` here again.

**Comment by abbysmal** [comment]
Thank you!
We will add back `--build-test` when the new version of ocaml-x509 is released then, merging. :)

**Pull request merged**